TensorFlowSharp/OperationsExtras.cs

```diff
@@ -31,7 +31,11 @@ TFOutput ReduceDims (TFOutput input, TFOutput? axis = null)
 			if (shape.IsFullySpecified) {
 				// The python code distinguishes between tensor and sparsetensor
 
-				return this.Const (shape.ToIntArray (), TFDataType.Int32);
+				var array = new int [shape.NumDimensions];
+				for (int i = 0; i < array.Length; i++)
+					array [i] = i;
+
+				return this.Const (array, TFDataType.Int32);
```

Comment:
I would appreciate @migueldeicaza and @cesarsouza to review this change. I returned the initial implementation of ReduceDims(), because changes introduced in [this ](https://github.com/migueldeicaza/TensorFlowSharp/commit/440066542a35fa62a013e5767b5d7f7ee47af56d#diff-9b54c6b41f47f0f34e4cbc0691cf1818) commit broke Should_ReduceMean() C# test.

My concern is whether we need to move this implementation to Shape.ToIntArray() method or leave here

Reply:
Hi @Dorokhov, 

So if I recall correctly, the first version of the code before the aforelinked commit was creating a vector of indices to indicate that every single dimension should be summed up such that the end result is a scalar, and the version after the commit is now returning the shape itself. I did not author this particular snippet, but _it seems_  to me that the previous version that you restored was the correct one, but I can be wrong though.

The tests for ReduceMean [came from the actual TensorFlow documentation](https://www.tensorflow.org/api_docs/python/tf/reduce_mean), so I would say they are still correct too, but again there could have been some hidden issue I didn't see.

If you ask me, I do not think this needs to be moved to Shape.ToIntArray() because, if I understood correctly, creating a vector of indices shouldn't be related to the task of converting a shape to int as done by Shape.ToIntArray(), but I guess that Miguel would be the best to advise here.

Regards,
Cesar

Reply:
This looks correct.